flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/typeutils/LegacyDataViewUtils.scala

```diff
@@ -200,12 +199,7 @@ object LegacyDataViewUtils {
           case _ =>
             instance.asInstanceOf[ListView[_]]
         }
-        val newTypeInfo = if (listView != null && listView.elementType != null) {
-          // use explicit element type if user has defined
-          new ListViewTypeInfo(listView.elementType)
-        } else {
-          list
-        }
+        val newTypeInfo = list
```

Comment:
Nit: It seems that `listView` can also be removed, and the variable `newTypeInfo` is not needed either.